### ISSUE TITLE
Instability of entmax bisect around 0?

### ISSUE THREAD

**Issue by azhx** [deep-spin/entmax#36]
Hi, I was plotting the output of the entmax_bisect function at different levels of alpha

```python
from matplotlib import pyplot as plt
x = torch.linspace(-1, 1, 10000).unsqueeze(-1)
x = torch.cat((x, torch.zeros_like(x)), dim=-1)
y = entmax_bisect(x, 10, dim=-1)[..., 0]

plt.plot(x.numpy(), y.numpy())
```

Here's what I'm getting
![image](https://github.com/deep-spin/entmax/assets/29048249/4049ed2e-6284-4aa2-ad36-508e0a661cd6)

I'm expecting this to be a smooth function except for two points when it transitions to 0 or 1

Any explanation for this?

**Comment by bpopeters** [comment]
Hello,

The issue seems to be about numerical stability (I can tell because the plot looks closer to expected when I switched to float64). We haven't really optimized for very large alpha values (10 is an _extremely_ large alpha value) because mathematically it becomes almost an expensive-to-compute step function/argmax. Do you have an application in mind for which you need this?

**Comment by TangLuo831** [comment]
hello,I use entmax_bisect(x, 1.33, dim=-1),get alpha<0.do you know why?
